hazelcast/src/main/java/com/hazelcast/map/impl/MapStoreWrapper.java

```diff
@@ -99,7 +100,8 @@ public void deleteAll(Collection keys) {
 
     public Iterable loadAllKeys() {
         if (isMapLoader()) {
-            return mapLoader.loadAllKeys();
+            // Invoke reflectively to preserve backwards binary compatibility
+            return ReflectionHelper.invokeMethod(mapLoader, "loadAllKeys");
```

Comment:
it should be just a fallback. 
calling a method reflectively (setAccessible(), etc) can cause funky issues when eg. SecurityManager is present.